core/src/main/scala/org/bitcoins/core/wallet/builder/RawTxFinalizer.scala

```diff
@@ -227,7 +227,8 @@ case class AddWitnessDataFinalizer(inputInfos: Vector[InputInfo])
 case class StandardNonInteractiveFinalizer(
     inputInfos: Vector[InputInfo],
     feeRate: FeeUnit,
-    changeSPK: ScriptPubKey)
+    changeSPK: ScriptPubKey,
+    otherFinalizerOpt: Option[RawTxFinalizer] = None)
```

Comment:
This seems like a hack to me, since we are still pre-0.4 do we want to change design to support this?

Reply:
Yeah, wasn't sure on the best way to do this. I'm guessing @nkohen will have the strongest opinion on it.

Reply:
So how we move forward here depends on the answer to the question
```
Should the "standard" finalizer shuffle or not?
```
Regardless of the answer to this question there should be two separate finalizers: one which does the shuffling and one which does not. The only question is which should be called standard and which should be called `[Shuffled | Unshuffled]NonInteractiveFinalizer`. Due to the add witness component of each composition won't be possible but this is not really an issue as it just means that there will be 10ish lines of code duplicated, none of which is logic (all construction).

TLDR: Get rid of the Option and make two different finalizers, the "standard" one gets to keep the companion object.

P.S. My vote is to leave standard as is and introduce a new one called shuffled until the PR that actually changes how we handle things internally.